### PR TITLE
Create tests for adding a configurable product to the shopping list

### PR DESCRIPTION
**Pull request by yuliia-mishchan**
## PR Description

Created tests for adding a configurable product to the shopping list, 
updating the preferred time of the day and preferred date of the configurable product in the shopping list.

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Review by AntonShopin — COMMENTED** [review]
Those tests are not enough. Please add the following tests and assertions:

* Test with not configured product in the SL -> Check that in the response it's not configured
* Configure 'not configured' product in the SL from the SL -> check that now the configuration is complete and all params are present
* Change the quantity of the Configured Product so Volume price is applied 
* Add 2 times Configurable product but with different configurations
* Add negative tests, like with the wrong "Preferred time of the day" or Date

In your tests, you don't check the result using GET requests, why? Testing the status code and SKU of the product is not enough, in the GET response there is info regarding configurable products

Also please resolve the conflicts

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 1]
```diff
@@ -458,4 +458,65 @@ Remove_a_concrete_product_from_the_shared_shopping_list_without_write_access_per
...
+#     And Response header parameter should be:    Content-Type    ${default_header_content_type}
+#     And Response status code should be:    422
+#     And Response should return error code:    901
+#     And Response reason should be:    Unprocessable Content
+#   [Teardown]    Run Keywords    I send a DELETE request:    /shopping-lists/${shoppingListId}
+#    ...    AND    Response status code should be:    204
+#    ...    AND    Response reason should be:    No Content
```
Comment (AntonShopin) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:477]: 1. Why this test is commented on? For the first mentioned bug we have a resolution, the second one - looks like not related to this test.
2. Am I right that the item (product) is not added to the SL in case invalid time of the day? If yes, please add an assertion that the product (item) was not added (is not present in the SL)

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 2]
```diff
@@ -458,4 +458,65 @@ Remove_a_concrete_product_from_the_shared_shopping_list_without_write_access_per
...
```
Comment (AntonShopin) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:492]: 1. Why this test is commented?
2. Am I right that the item (product) is not added to the SL in case date is missing? If yes, please add an assertion that the product (item) was not added (is not present in the SL)
3. Shoul API return a message in the response, that the date is not provided?

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 3]
```diff
@@ -458,4 +458,65 @@ Remove_a_concrete_product_from_the_shared_shopping_list_without_write_access_per
...
```
Comment (AntonShopin) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:507]: Why this test is commented? The related bug looks like already has a resolution

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot [hunk 4]
```diff
@@ -458,4 +458,65 @@ Remove_a_concrete_product_from_the_shared_shopping_list_without_write_access_per
...
```
Comment (AntonShopin) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot:522]: Can't approve until it's comented

+ what about other negative cases, like: 
* not provided quantity, quantity in a string instead of integer
* same for `availableQuantity`
* same for `isComplete`
* and so on

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/positive.robot [hunk 5]
```diff
@@ -261,4 +261,146 @@ Remove_a_concrete_product_from_the_shared_shopping_list
...
+    And Response status code should be:    201
+    And Response body parameter should be:    [data][attributes][quantity]    3
+    And Response body parameter should be:    [data][attributes][sku]    ${configurable_product.sku}
+    And Response body parameter should be:    [data][attributes][productConfigurationInstance][displayData]  {"Preferred time of the day":"Afternoon","Date":"9.09.2050"} 
+    And Response body parameter should be:    [data][attributes][productConfigurationInstance][isComplete]  True
```
Comment (AntonShopin) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/positive.robot:277]: Don't we need to make a GET request and check that the item is there and the configuration is present? 
Like `GET /shopping-lists/${shoppingListId}` or `GET /shopping-lists/${shoppingListId}?include=shopping-list-items,concrete-products`

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/positive.robot [hunk 6]
```diff
@@ -261,4 +261,146 @@ Remove_a_concrete_product_from_the_shared_shopping_list
...
+   And Response body parameter should be:    [included][0][attributes][productConfigurationInstance][isComplete]    True
+   And Response body parameter should be:   [included][1][id]    ${shoppingListItemId2}
+   And Response body parameter should be:   [included][1][attributes][quantity]    1
+   And Response body parameter should be:   [included][1][attributes][sku]    ${configurable_product.sku}
+   And Response body parameter should be:    [included][1][attributes][productConfigurationInstance][displayData]    {\"Preferred time of the day\":\"Afternoon\",\"Date\":\"11.11.2040\"}
+   And Response body parameter should be:    [included][1][attributes][productConfigurationInstance][isComplete]    False
+   And Response include element has self link:   shopping-list-items
+   [Teardown]    Run Keywords    I send a DELETE request:    /shopping-lists/${ShoppingListId}
+    ...    AND    Response status code should be:    204
+    ...    AND    Response reason should be:    No Content
```
Comment (AntonShopin) [inline tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/positive.robot:406]: Could you please add one more test, when we add to the SL regular product and Configurable product. 
TIA

File: tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot (hunk 1 above)
Comment (yuliia-mishchan) [reply tests/api/suite_nonsplit/glue/shopping_list_endpoints/shopping_list_items/negative.robot]: Spryker provides an Example configuration. Only available quantity and price could somehow influence concrete products.
Other properties in configuration (that Spryker did for Example configuration) it’s just visible/selectable values that the customer could change.

**Comment by AntonShopin** [comment]
- please add to at least one negative test an assertion, that CP was NOT added to the Shopping List
- Unify assertions. In some cases you use `And Array in response should contain property with value:    [errors]    detail    quantity => This value should be greater than 0.` and in other `And Response should return error message:    quantity => This value should be greater than 0.`
- rename first negative test `Add_a_regular_product_to_the_shopping_list_with_configuration` -> it sounds like a positive test
- add test for deleting Configurable Product from the Shopping list
- please check and add a test with Configurable Product (if possible): is it possible to add product from the SL to the Shopping Cart
